src/engine/ruleset/decoders/wazuh-agent/upgrade/upgrade.yml

```diff
@@ -0,0 +1,53 @@
+---
+name: decoder/upgrade/0
+
+metadata:
+    module: wazuh-agent/upgrade
+    title: Upgrade Confirmation events
+    description: >
+        Decodes Upgrade confirmation messages
+    compatibility: >
+        This decoder has been tested on Wazuh version 4.4.0
+    versions:
+      - "4.4.0"
+    author:
+        name: Wazuh, Inc.
+        url: https://wazuh.com
+        date: 2023/03/22
+    references:
+        - https://documentation.wazuh.com/current/user-manual/agents/remote-upgrading/upgrading-agent.html#upgrading-agent
+        - https://documentation.wazuh.com/current/user-manual/agents/remote-upgrading/agent-upgrade-module.html
+
+sources:
+  - decoder/queue-upgrade/0
+
+check:
+  - event.original: +starts_with/{
+
+# Examples:
+# https://github.com/wazuh/wazuh/issues/5406
+# {"command":"upgrade_update_status","parameters":{"error":0,"message":"Upgrade was successful","status":"Done"}}
+# {"command":"upgrade_update_status","parameters":{"error":2,"message":"Upgrade Failed","status":"Failed"}}
+
+parse:
+  logpar:
+    - event.original: <~json/json>
```

Comment:
Is it possible that an event coming from the upgrade queue is not a Json?

Reply:
While testing I didn't receive a non-json event and didn't found on the wazuh code another type of message. I guess we could delete the check without any impact.